### PR TITLE
TNG-777 initial documentation and test commit

### PR DESCRIPTION
**Pull request by vlapl**

### CODE REVIEW

File: src/components/Dropdown/__tests__/Dropdown.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          { key: "opt3", label: "Option 3", path: "path3" },
+        ],
+      },
+    });
+
+    const option1 = getByTestId("option-opt1");
+
+    expect(option1.innerHTML).toEqual("Option 1");
+    await fireEvent(option1!, new Event("click"));
+    expect(spy).toHaveBeenCalled();
```
Comment (tgolbs) [inline src/components/Dropdown/__tests__/Dropdown.spec.tsx:24]: Please check that the correct item is handed to the callback.

Comment (tgolbs) [reply src/components/Dropdown/__tests__/Dropdown.spec.tsx:24]: You can access the parameters of the mocked call through:

`spy.mock.calls[0]`

File: src/components/Dropdown/__tests__/Dropdown.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  });
+
+  it("should render as many options as are provided", async () => {
```
Comment (tgolbs) [inline src/components/Dropdown/__tests__/Dropdown.spec.tsx:27]: Please change this test functionality that it loops over the provided options and checks through `getByText` that the option was rendered.

File: src/documentation/examples/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  { key: "opt2", label: "Option_2", path: "path2" },
+];
+
+const handleClick = (item: Option) => {
+  console.log(item.label);
+};
+
+@Component
+export default class App extends Vue {
+  currentItem: any = null;
```
Comment (tgolbs) [inline src/documentation/examples/Dropdown.tsx:18]: Either remove the local currentItem state or use it in your example.

**Review by tgolbs — CHANGES_REQUESTED** [review]

File: src/components/Dropdown/__tests__/Dropdown.spec.tsx (hunk 1 above)
Comment (vlapl) [reply src/components/Dropdown/__tests__/Dropdown.spec.tsx:25]: The following line is checking this:

File: src/documentation/examples/Dropdown.tsx (hunk 3 above)
Comment (vlapl) [reply src/documentation/examples/Dropdown.tsx:18]: Removed.

**Review by tgolbs — APPROVED** [review]

**Pull request merged**